src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -292,6 +300,45 @@ public void onBlockBreak(BlockBreakEvent e) {
         }
     }
 
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() > 0 && Talisman.checkFor(e, SlimefunItems.TALISMAN_WISE)) {
+            e.setAmount(e.getAmount() * 2);
+        }
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity();
+        Location respawnLocation = getSafeRespawnLocation(player);
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+
+        if (dmgCause == DamageCause.VOID && Talisman.checkFor(e, SlimefunItems.TALISMAN_RESURRECTED)) {
+            SlimefunPlugin.runSync(() -> {
+                PaperLib.teleportAsync(player, respawnLocation);
+            }, 2);
```

Comment:
Why the delay here?

Reply:
If the player is dead, teleporting will fail, so I need to do a little delay so the teleport happens after the death event is cancelled

Reply:
But why not put the Player's UUID into a Set and then set the spawn location on the PlayerRespawnEvent?

Reply:
This seemed more convenient at the time